src/openfermion/ops/_interaction_tensor.py

```diff
@@ -84,103 +85,114 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
-            n_qubits x n_qubits x n_qubits x n_qubits numpy array offloats.
-            For instance, the two body term of MolecularOperator.
+        constant(complex): A constant term in the operator given as a complex
+            number. For instance, the nuclear repulsion energy.
+        n_body_tensors(dict): A dictionary storing the tensors describing
+            n-body interactions. For instance, n_body_tensors[2] is a
+            n_qubits x n_qubits x n_qubits x n_qubits numpy array of complex
+            numbers.
     """
 
-    def __init__(self, constant, one_body_tensor, two_body_tensor):
+    def __init__(self, constant, *args, n_body_tensors=None, n_qubits=None):
         """Initialize the InteractionTensor class.
 
         Args:
-            constant: A constant term in the operator given as a
-                float. For instance, the nuclear repulsion energy.
-            one_body_tensor: The coefficients of the 2D matrix terms. This
-                is an n_qubits x n_qubits numpy array of floats. For
-                instance, the one body term of MolecularOperator.
-            two_body_tensor: The coefficients of the 4D matrix terms. This is
-                an n_qubits x n_qubits x n_qubits x n_qubits numpy array of
-                floats. For instance, the two body term of MolecularOperator.
+            constant(complex): A constant term in the operator given as a
+                complex number. For instance, the nuclear repulsion energy.
+            n_body_tensors(dict): A dictionary storing the tensors describing
+                n-body interactions. For instance, n_body_tensors[2] is a
+                n_qubits x n_qubits x n_qubits x n_qubits numpy array of
+                complex numbers.
         """
-        if constant is None:
-            constant = 0.0
-        self.constant = constant
-        self.one_body_tensor = one_body_tensor
-        self.two_body_tensor = two_body_tensor
-        self.n_qubits = self.one_body_tensor.shape[0]
+        # initialize constant
+        if constant is not None:
+            self.constant = constant
+        else:
+            self.constant = 0.
+
+        # initialize n_body_tensors
+        if args:
+            # tensors were passed in as arguments
+            self.n_body_tensors = dict()
+            for i in range(len(args)):
+                self.n_body_tensors[i + 1] = args[i]
+        elif n_body_tensors:
+            # tensors were passed in directly as dictionary
+            self.n_body_tensors = n_body_tensors
+        else:
+            # no tensors were given
+            self.n_body_tensors = dict()
+
+        # initialize n_qubits
+        if n_qubits:
+            self.n_qubits = n_qubits
+        elif 1 in self.n_body_tensors:
+            self.n_qubits = self.n_body_tensors[1].shape[0]
+        else:
+            raise ValueError("Could not determine n_qubits.")
 
     def __getitem__(self, args):
```

Comment:
I'm not sure I understand. I think args represents a list of indices, right? So it can be any list of integers of any length. Indeed, the tests for the getitem method all passed.

Reply:
We really need to all get on the same page about the exact keys of this dictionary before having this discussion.